libloki/test/libloki-protocol_test.js

```diff
@@ -0,0 +1,95 @@
+/* global libsignal, libloki, textsecure, StringView */
+
+'use strict';
+
+describe('FallBackSessionCipher', () => {
+  let fallbackCipher;
+  let identityKey;
+  let address;
+  const store = textsecure.storage.protocol;
+
+  before(async () => {
+    clearDatabase();
+    identityKey = await libsignal.KeyHelper.generateIdentityKeyPair();
+    store.put('identityKey', identityKey);
+    const key = libsignal.crypto.getRandomBytes(32);
+    const pubKeyString = StringView.arrayBufferToHex(key);
+    address = new libsignal.SignalProtocolAddress(
+      pubKeyString,
+      1
+    );
+    fallbackCipher = new libloki.FallBackSessionCipher(address);
+  });
+
+  it('should encrypt fallback cipher messages as friend requests', async () => {
+    const buffer = new ArrayBuffer(10);
+    const { type } = await fallbackCipher.encrypt(buffer);
+    assert.strictEqual(type, textsecure.protobuf.Envelope.Type.FRIEND_REQUEST);
+  });
```

Comment:
Would be good to actually test if the encryption worked by decrypting the message and checking if it matches the initial buffer.

Reply:
Whoops lol thought I did that